flaky tests: docker-py: ServiceTest.test_create_service_with_secret

Seen failing frequently;

```
09:33:46 =================================== FAILURES ===================================
09:33:46 _________________ ServiceTest.test_create_service_with_secret __________________
09:33:46 /docker-py/tests/integration/api_service_test.py:429: in test_create_service_with_secret
09:33:46     container = self.get_service_container(name)
09:33:46 /docker-py/tests/integration/api_service_test.py:41: in get_service_container
09:33:46     all=include_stopped
09:33:46 /docker-py/docker/api/container.py:189: in containers
09:33:46     res = self._result(self._get(u, params=params), True)
09:33:46 /docker-py/docker/utils/decorators.py:46: in inner
09:33:46     return f(self, *args, **kwargs)
09:33:46 /docker-py/docker/api/client.py:183: in _get
09:33:46     return self.get(url, **self._set_request_timeout(kwargs))
09:33:46 /usr/lib/python2.7/dist-packages/requests/sessions.py:469: in get
09:33:46     return self.request('GET', url, **kwargs)
09:33:46 /usr/lib/python2.7/dist-packages/requests/sessions.py:457: in request
09:33:46     resp = self.send(prep, **send_kwargs)
09:33:46 /usr/lib/python2.7/dist-packages/requests/sessions.py:569: in send
09:33:46     r = adapter.send(request, **kwargs)
09:33:46 /usr/lib/python2.7/dist-packages/requests/adapters.py:407: in send
09:33:46     raise ConnectionError(err, request=request)
09:33:46 E   ConnectionError: ('Connection aborted.', BadStatusLine("''",))
09:33:46 ========= 1 failed, 243 passed, 9 skipped, 3 xfailed in 391.56 seconds =========
```

e.g., https://jenkins.dockerproject.org/job/Docker-PRs/43957/console (https://github.com/moby/moby/pull/33860)

#33882 avoids the panic and causes the test to fail in this way instead:

```
/docker-py/tests/integration/api_service_test.py:454: in test_create_service_with_unicode_secret
    container = self.get_service_container(name)
/docker-py/tests/integration/api_service_test.py:41: in get_service_container
    all=include_stopped
/docker-py/docker/api/container.py:189: in containers
    res = self._result(self._get(u, params=params), True)
/docker-py/docker/api/client.py:220: in _result
    self._raise_for_status(response)
/docker-py/docker/api/client.py:216: in _raise_for_status
    raise create_api_error_from_http_exception(e)
/docker-py/docker/errors.py:31: in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
E   NotFound: 404 Client Error: Not Found ("no such container 050224bd46a04143132fac672514ae4f4c4b2e7147c6a8a4757ecc271c427f9f")
```

Not a complete solution, since the test is still failing, but hopefully this will help track down the actual problem.

--- a/daemon/list.go
+++ b/daemon/list.go
@@ -162,11 +162,13 @@ func (daemon *Daemon) filterByNameIDMatches(view container.View, ctx *listContex
 	cntrs := make([]container.Snapshot, 0, len(matches))
 	for id := range matches {
 		c, err := view.Get(id)
-		if err != nil {
-			return nil, err
-		}
-		if c != nil {
+		switch err.(type) {
+		case nil:
 			cntrs = append(cntrs, *c)
+		case container.NoSuchContainerError:
+			// ignore error
+		default:
+			return nil, err
 		}
 	}
 

